idaes/models/unit_models/stream_scaler.py

```diff
@@ -0,0 +1,255 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+Unit model to adjust size of streams to represent, for example, a stream being split across several identical units,
+which are then all modeled as a single IDAES unit
+"""
+from enum import Enum
+from functools import partial
+
+from pyomo.environ import (
+    Block,
+    check_optimal_termination,
+    Param,
+    PositiveReals,
+    Reals,
+    RangeSet,
+    units as pyunits,
+    Var,
+)
+from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In, ListOf, Bool
+
+from idaes.core import (
+    declare_process_block_class,
+    UnitModelBlockData,
+    useDefault,
+    MaterialBalanceType,
+    MaterialFlowBasis,
+)
+from idaes.core.util.config import (
+    is_physical_parameter_block,
+    is_state_block,
+)
+from idaes.core.util.exceptions import (
+    BurntToast,
+    ConfigurationError,
+    PropertyNotSupportedError,
+    InitializationError,
+)
+from idaes.core.base.var_like_expression import VarLikeExpression
+from idaes.core.util.math import smooth_min
+from idaes.core.util.tables import create_stream_table_dataframe
+import idaes.core.util.scaling as iscale
+from idaes.core.solvers import get_solver
+import idaes.logger as idaeslog
+
+from idaes.models.unit_models.feed import FeedInitializer as StreamScalerInitializer
+
+__author__ = "Douglas Allan"
```

Comment:
Should @tannerpolley be one of the authors here?

Reply:
I actually didnt build the actual unit model itself and didn't edit any of the code here. I was just asked to implement the integration of the unit model into the main repo